packages/core/src/extensions/extension-loader/file-system-provisioner-store/ensure-hashed-directory-for-extension.injectable.ts

```diff
@@ -24,8 +25,15 @@ const ensureHashedDirectoryForExtensionInjectable = getInjectable({
     const directoryForExtensionData = di.inject(directoryForExtensionDataInjectable);
     const ensureDirectory = di.inject(ensureDirInjectable);
     const getHash = di.inject(getHashInjectable);
+    const findKeyContains = di.inject(findKeyContainsInjectable)<string>;
 
     return async (extensionName, registeredExtensions) => {
+      const legacyDirPath = findKeyContains(registeredExtensions, extensionName);
+
+      if (legacyDirPath) {
+        return legacyDirPath;
```

Comment:
instead of returning early here, we should still make sure that `ensureDir` is run after this

Reply:
done